conda_forge_tick/auto_tick.py

```diff
@@ -213,20 +264,25 @@ def run(
         is_version=is_version_migration,
     )
 
-    # TODO: run this in parallel
-    repo = get_repo(context=context, branch=branch_name, base_branch=base_branch)
-
-    feedstock_dir = str(context.local_clone_dir)
-    if not feedstock_dir or not repo:
-        logger.critical(
-            "Failed to migrate %s, %s",
-            context.feedstock_name,
-            context.attrs.get("pr_info", {}).get("bad"),
-        )
+    branch_name = migrator.remote_branch(context) + "_h" + uuid4().hex[0:6]
+    if not _prepare_feedstock_repository(
+        git_backend,
+        context,
+        branch_name,
+        base_branch,
+    ):
+        # something went wrong during forking or cloning
         return False, False
 
     # need to use an absolute path here
-    feedstock_dir = os.path.abspath(feedstock_dir)
+    feedstock_dir = str(context.local_clone_dir.resolve())
+
+    # This is needed because we want to migrate to the new backend step-by-step
+    repo: github3.repos.Repository | None = github3_client().repository(
+        context.git_repo_owner, context.git_repo_name
+    )
+
+    assert repo is not None
```

Comment:
I don't think we have a bug here. If a NotFoundError is raised, it is not caught in `run,` which is our desired behavior. The `assert` is there because I saw in Github3s source code that None can be returned without error in some weird API edge cases: [source code](https://github.com/sigmavirus24/github3.py/blob/f9f534b211b0ddce41e3772b89091d49c748e5e2/src/github3/github.py#L2104)

Reply:
Ahh ok. That code path only happens if the repo owner and name are both `None`.

Reply:
Yeah I think when that happens we want the assert to be there